### PR TITLE
chore: more core cleanup,  lock upgrade

### PR DESCRIPTION
**Pull request by nicolasburtey**
this refactoring fix a few things:
- removing one more file from /core folder
- the new redlock have now a `using` function which has a automaticExtensionThreshold function. this is super handy because now we don't need to do a new check within a critical section to check if the redlock is still active: the lock should be automatically extended, and if not, `signal` will be aborted.
- because of the former point, we use to need a re-entrant lock. now this seems no longer any more.

### CODE REVIEW

File: src/services/lock/index.ts [hunk 1]
```diff
@@ -1,76 +1,101 @@
...
 export const LockService = (): ILockService => {
   const lockWalletId = async <Res>(
-    {
-      walletId,
-      logger,
-      lock,
-    }: { walletId: WalletId; logger: Logger; lock?: DistributedLock },
-    f: () => Promise<Res>,
+    { walletId }: { walletId: WalletId },
+    f: (signal: WalletIdAbortSignal) => Promise<Res>,
```
Comment (dolcalmi) [inline src/services/lock/index.ts:76]: any reason to not use 1 param rule? (same apply to lockPaymentHash and lockOnChainTxHash)

File: package.json [hunk 2]
```diff
@@ -97,7 +96,7 @@
     "pino-http": "^6.6.0",
     "prom-client": "^13.1.0",
     "rate-limiter-flexible": "^2.2.2",
-    "redlock": "^4.2.0",
+    "redlock": "^5.0.0-beta.2",
```
Comment (dolcalmi) [inline package.json:99]: do you think is safe to use a beta version? also please remove yarn.lock and create it again

File: src/services/lock/index.ts [hunk 3]
```diff
@@ -1,76 +1,101 @@
...
+export const redlock = async ({ path, signal }: IRedLock, async_fn) => {
+  if (signal) {
+    if (signal.aborted) {
+      return new ResourceExpiredLockServiceError(signal.error?.message)
+    }
+    return async_fn(signal)
+  }
+
+  try {
+    return await redlockClient.using([path], ttl, async (signal) => async_fn(signal))
```
Comment (dolcalmi) [inline src/services/lock/index.ts:63]:
```suggestion
    return await redlockClient.using([path], ttl, async_fn) // avoid snake case 
```

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (nicolasburtey) [reply package.json:99]: yes I think so: the beta 2 was published 3 months ago, and the package has 250k download weekly/1m monthly.

File: src/services/lock/index.ts [hunk 4]
```diff
@@ -1,76 +1,101 @@
...
+    // attempted with the `using` API.
+    automaticExtensionThreshold: 2500, // time in ms
+  },
+)
+
+const getWalletLockResource = (path) => `locks:wallet:${path}`
+const getPaymentHashLockResource = (path) => `locks:paymenthash:${path}`
+const getOnChainTxHashLockResource = (path) => `locks:onchaintxhash:${path}`
+
+interface IRedLock {
```
[the remark this answers is not in the document]
Comment (nicolasburtey) [reply src/services/lock/index.ts:49]: what would you use instead?

File: src/services/lock/index.ts [hunk 5]
```diff
@@ -1,76 +1,101 @@
...
+  path: string
+  signal?: RedlockAbortSignal
+}
+
+export const redlock = async ({ path, signal }: IRedLock, async_fn) => {
```
[the remark this answers is not in the document]
Comment (nicolasburtey) [reply src/services/lock/index.ts:54]: done for snake case. for asyncFn, I'm not sure how to type it, it's currently `implicit any`.

File: src/services/lock/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dolcalmi) [reply src/services/lock/index.ts]: just a Type

File: src/services/lock/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (dolcalmi) [reply src/services/lock/index.ts]: probably the same type used by redlock using
```js
using<T>(resources: string[], duration: number, routine: (signal: RedlockAbortSignal) => Promise<T>): Promise<T>
```

redlock args type:
<details>
  <summary>Click to expand!</summary>

  ```js
    type RedlockArgs = {
      path: string
      signal?: RedlockAbortSignal
      asyncFn: <T>(signal: RedlockAbortSignal) => Promise<T>
    }
  ```
</details>

redlock:
<details>
<summary>Click to expand!</summary>

```js
  export const redlock = async ({ path, signal, asyncFn }: RedlockArgs) => {
    if (signal) {
      if (signal.aborted) {
        return new ResourceExpiredLockServiceError(signal.error?.message)
      }
      return asyncFn(signal)
    }

    try {
      return await redlockClient.using([path], ttl, asyncFn)
    } catch (error) {
      if (error instanceof ExecutionError) {
        return new ResourceAttemptsLockServiceError()
      }

      return new UnknownLockServiceError()
    }
  }
```
</details>

**Pull request merged**